sale_timesheet_rounded/models/account_analytic_line.py

```diff
@@ -0,0 +1,166 @@
+# Copyright 2019 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+import logging
+
+from odoo import api, fields, models
+from odoo.tools.float_utils import float_round
+
+_logger = logging.getLogger(__name__)
+
+
+class AccountAnalyticLine(models.Model):
+
+    _inherit = "account.analytic.line"
+
+    unit_amount_rounded = fields.Float(
+        string="Quantity rounded",
+        # Important: do NOT pass a default here.
+        # Passing a default will break computation at create
+        # since we want to allow to pass specific values
+        # we must check if the value is there or not.
+        # If you set a default, you are going to get it as value at create
+        # and the computation won't happen.
+        # default=0.0,
+        copy=False,
+    )
+
+    @api.model
+    def _init_column(self, column_name):
+        """ Initialize the value of the given column for existing rows.
+            Overridden here because we need to have different default values
+            for unit_amount_rounded for every analytic line.
+        """
+        if column_name != "unit_amount_rounded":
+            super()._init_column(column_name)
+        else:
+            _logger.info(
+                "Initializing column `unit_amount_rounded` with the "
+                "value of `unit_amount`"
+            )
+            query = 'UPDATE "%s" SET unit_amount_rounded = unit_amount;' % self._table
+            self.env.cr.execute(query)
+
+    @api.onchange("unit_amount")
+    def _onchange_unit_amount(self):
+        self.unit_amount_rounded = self.with_context(
+            force_compute=1
+        )._calc_unit_amount_rounded()
+
+    def _calc_unit_amount_rounded(self):
+        self.ensure_one()
+        force_compute = self.env.context.get("force_compute")
+        # TODO: do we still need this?
```

Comment:
Can you check this TODO ?

Reply:
I don't know as it not completely covered with tests, maybe after internal test, we will be able to improve it